core/src/main/java/nu/yona/server/subscriptions/service/UserService.java

```diff
@@ -751,10 +753,40 @@ public static String getPassword(Optional<String> password)
 	 */
 	@Transactional
 	public User getUserEntityById(UUID id)
+	{
+		return getUserEntityById(id, LockModeType.NONE);
+	}
+
+	/**
+	 * This method returns a user entity. The passed on Id is checked whether or not it is set. it also checks that the return
+	 * value is always the user entity. If not an exception is thrown. A pessimistic database write lock is claimed when fetching
+	 * the entity.
+	 * 
+	 * @param id the ID of the user
+	 * @return The user entity (never null)
+	 */
+	@Transactional
+	private User getUserEntityByIdWithUpdateLock(UUID id)
```

Comment:
Does the `@Transactional` annotation make sense on a private method?

Reply:
It doesn't, so I've removed it. The method is taken from an old version, where it was public. I made it private but forgot to remove the annotation.